backend/civic_intelligence.py

```
@@ -0,0 +1,246 @@
+            },
+            "top_concern": trends["top_keywords"][0][0] if trends["top_keywords"] else "None"
+        }
+
+    def _save_snapshot(self, data: Dict[str, Any]):
+        filename = f"{datetime.now().strftime('%Y-%m-%d')}.json"
+        filepath = os.path.join(self.snapshot_dir, filename)
+        try:
+            with open(filepath, 'w') as f:
+                json.dump(data, f, indent=4)
+            logger.info(f"Saved daily snapshot to {filepath}")
+        except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)